src/module.cpp

```diff
@@ -120,6 +122,27 @@ void module::assign(const module& m)
         }
         else
         {
+            // if there are sub_module inputs, need to make a copy of the submodule
```

Comment:
The module class should not copy the submodules. When copying a module the module_refs should be the same.

Reply:
When should we replace the old_mod_ref with the new_mod_ref?

Reply:
The program would replace the refs since it can see all modules.

Reply:
fixed.